### ISSUE TITLE
@slash_command after reloading an extension no longer works.

### ISSUE THREAD

**Issue by itsflorent** [Pycord-Development/pycord#477]
### Summary

@slash_command reloading extension bug

### Reproduction Steps

I used bot.load_extension("commands.slash"), after reloading the slash command in the cog doesn't work anymore.

### Minimal Reproducible Code

```python
@bot.command()
async def reload(ctx):
    msg = await ctx.reply("**Reloading all cogs...**")

    bot.unload_extension("commands.slash")
    bot.load_extension("commands.slash")

    await msg.edit(content=f"**Finish reloading all cogs!**")
```

### Expected Results

That the slash commands work again after reloading.

### Actual Results

![image](https://user-images.githubusercontent.com/70605487/142697367-9739ba14-f1dc-4893-893a-1d9cd1380dde.png)

### Intents

all()

### System Information

![image](https://user-images.githubusercontent.com/70605487/142697440-9fcb2366-3fd0-4f1e-be5b-f8987e9af0ad.png)

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

**Comment by oskiwii** [comment]
What does it say if you enable logging (set to debug maybe)?

**Comment by BobDotCom** [comment]
Expected and will be fixed in #433
